schema/string_test.go

```diff
@@ -16,44 +16,44 @@ func TestDecodeString_InvalidUUIDVersion(t *testing.T) {
 	}
 }
 
-func TestDecodeString_Success(t *testing.T) {
-	t.Run("ValidMinLength", func(t *testing.T) {
-		var data = []struct {
-			Desc        string
-			Value       string
-			Format      string
-			constraints Constraints
-		}{
-			{"URI", "http://google.com", stringURI, Constraints{MinLength: 5}},
-			{"Email", "foo@bar.com", stringEmail, Constraints{MinLength: 3}},
-			{"UUID", "C56A4180-65AA-42EC-A945-5FD21DEC0538", stringUUID, Constraints{MinLength: 36}},
-		}
-		for _, d := range data {
-			v, err := decodeString(d.Format, d.Value, d.constraints)
-			if err != nil {
-				t.Errorf("want:nil got:%q", err)
-			}
-			if v != d.Value {
-				t.Errorf("want:%s got:%s", d.Value, v)
+func TestDecodeString_ErrorCheckingConstraints(t *testing.T) {
+	data := []struct {
+		desc        string
+		value       string
+		format      string
+		constraints Constraints
+	}{
+		{"InvalidMinLength", "6fa459ea-ee8a-3ca4-894e-db77e160355e", stringUUID, Constraints{MinLength: 100}},
+		{"InvalidMinLength", "foo@bar.com", stringEmail, Constraints{MinLength: 100}},
+		{"InvalidMinLength", "http://google.com", stringURI, Constraints{MinLength: 100}},
+	}
+	for _, d := range data {
+		t.Run(d.desc, func(t *testing.T) {
+			if _, err := decodeString(d.format, d.value, d.constraints); err == nil {
+				t.Fatalf("err want:err got:nil")
 			}
-		}
-	})
+		})
+	}
+}
 
-	t.Run("Error", func(t *testing.T) {
-		var data = []struct {
-			Desc        string
-			Value       string
-			Format      string
-			constraints Constraints
-		}{
-			{"InvalidMinLengthUUID", "C56A4180-65AA-42EC-A945-5FD21DEC0538", stringUUID, Constraints{MinLength: 40}},
+func TestDecodeString_Success(t *testing.T) {
+	var data = []struct {
+		Desc        string
+		Value       string
+		Format      string
+		constraints Constraints
+	}{
+		{"URI", "http://google.com", stringURI, Constraints{}},
+		{"Email", "foo@bar.com", stringEmail, Constraints{}},
+		{"UUID", "C56A4180-65AA-42EC-A945-5FD21DEC0538", stringUUID, Constraints{}},
+	}
+	for _, d := range data {
+		v, err := decodeString(d.Format, d.Value, d.constraints)
```

Comment:
Add subtests? Just like the test function above?

Reply:
Let me check again.
I can follow this [function](https://github.com/frictionlessdata/tableschema-go/blob/master/schema/time_test.go#L8) to create three subtests in function `TestDecodeString_ErrorCheckingConstraints`.
And those three subtests contains `InvalidMinLength`, `InvalidMaxLength` and `InvalidPattern` ?

Reply:
Maybe there is small confusion. This comment is about the success case, right?

My suggestion was  simply:

```go
+func TestDecodeString_Success(t *testing.T) {
  	var data = []struct {		  	var data = []struct {
 -		Desc   string		 +		Desc        string
 -		Value  string		 +		Value       string
 -		Format string		 +		Format      string
 +		constraints Constraints
  	}{		  	}{
                {"URI", "http://google.com", stringURI, Constraints{MinLength:1}},
                {"Email", "foo@bar.com", stringEmail, Constraints{MinLength:1}},
                {"UUID", "C56A4180-65AA-42EC-A945-5FD21DEC0538", stringUUID, Constraints{MinLength:1}},
  	}
  	for _, d := range data {
                t.Run(d.Desc, func(t *testing.T) {
                     v, err := decodeString(d.Format, d.Value, d.constraints)
       		     if err != nil {
  			     t.Errorf("want:nil got:%q", err)
      		     }
   		     if v != d.Value {
 			     t.Errorf("want:%s got:%s", d.Value, v)
 		     }
             })
 	}
 }
```

Then you're checking the success case but considering the constraints.

What do you think?